code/routers/user.js

```diff
@@ -0,0 +1,312 @@
+    {username: username},
+    {email: username}
+  ]}
+
+  User.findOne(user_query, function(err, user, next) {
+    if (user) {
+      return res.status(400).send({
+        message: 'User already exists with that username or email'
+      });
+    }else{
+      // create user
+      let newUser = new User();
+
+      newUser.username = username;
+      newUser.firstName = firstName;
+      newUser.lastName = lastName;
+      newUser.email = email;
+      newUser.setPassword(password);
+
+      newUser.save(function(err, user_check) {
+        if(err) {
+          return res.status(400).send({
+            message: 'Failed to add user'
+          });
+        }
+        else {
+          return res.status(201).send({
+            message: 'User added successfully'
+          });
+        }
+      });
+    }
+  });
+
+});
+
+
+/*
+  forgot password api
+
+  let's users reset their password and redirect
+  them to the login page
+*/
+
+router.get('/forgot', function(req, res) {
+  res.redirect('forgot');
+});
+
+router.post('/forgot', function(req, res, next) {
+  async.waterfall([
+    function(done) {
+      crypto.randomBytes(20, function(err, buf) {
+        var token = buf.toString('hex');
+        done(err, token);
+      });
+    },
+    function(token, done) {
+      User.findOne({ email: req.body.email }, function(err, user) {
+        if (!user) {
+          //req.flash('error', 'No account with that email address exists.');
+          //return res.redirect('/forgot');
+        }
+
+        user.resetPasswordToken = token;
+        user.resetPasswordExpires = Date.now() + 3600000; // 1 hour
+
+        user.save(function(err) {
+          done(err, token, user);
+        });
+      });
+    },
+    function(token, user, done) {
+      var smtpTransport = nodemailer.createTransport({
+        host: "smtp.gmail.com",
+        auth: {
+          type: "login",
+          user: "DOMAD24901@gmail.com",
+          pass: "DoMADTemp#2020",
+        }
+      });
+
+      var mailOptions = {
+        to: user.email,
+        from: 'DOMAD24901@gmail.com',
+        subject: 'Node.js Password Reset',
+        text: 'You are receiving this because you (or someone else) have requested the reset of the password for your account.\n\n' +
+          'Please click on the following link, or paste this into your browser to complete the process:\n\n' +
+          'http://localhost:' + 3000 + '/reset/' + token + '\n\n' +
+          'If you did not request this, please ignore this email and your password will remain unchanged.\n'
+      };
+      smtpTransport.sendMail(mailOptions, function(err) {
+        console.log('mail sent');
+        console.log('email: ' + user.email)
+        //req.flash('success', 'An e-mail has been sent to ' + user.email + ' with further instructions.');
+        done(err, 'done');
+      });
+    }
+  ], function(err) {
+    if (err) return next(err);
+    res.redirect('/forgot');
+  });
+});
+
+// :token -> check-token
```

Comment:
Do we still want to change this api call to `check-token` instead of `:token` to make it a little more clear?